src/main/java/uk/gov/hmcts/reform/idam/service/RestoreUserService.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.hmcts.reform.idam.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import feign.Response;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.idam.service.remote.client.IdamClient;
+import uk.gov.hmcts.reform.idam.service.remote.requests.RestoreUserRequest;
+import uk.gov.hmcts.reform.idam.service.remote.responses.DeletionLog;
+import uk.gov.hmcts.reform.idam.service.remote.responses.IdamCreateUserErrorResponse;
+import uk.gov.hmcts.reform.idam.util.IdamTokenGenerator;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static uk.gov.hmcts.reform.idam.service.IdamUserRestorerService.MARKER;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RestoreUserService {
+
+    private final IdamClient idamClient;
+    private final IdamTokenGenerator idamTokenGenerator;
+
+    public boolean restoreUser(DeletionLog deletionLog) {
+
+        RestoreUserRequest requestBody = createRequestBody(deletionLog);
+        HttpStatus status;
+        try {
+            status = callApi(deletionLog.getUserId(), requestBody);
+        } catch (IOException ioe) {
+            log.error(
+                "[{}] Failed to read idam response from restore user {}. Exception {}",
+                MARKER,
+                deletionLog.getUserId(),
+                ioe.getMessage(),
+                ioe
+            );
+            return false;
+        }
+
+        if (status == HttpStatus.CREATED) {
+            log.info("Successfully restored user with id {}", deletionLog.getUserId());
+        } else {
+            log.error(
+                "[{}] Failed to restore user from deletion log {}. HTTP Status {}",
+                MARKER,
+                deletionLog.getUserId(),
+                status
+            );
+        }
+        return status == HttpStatus.CREATED;
+    }
+
+    private HttpStatus callApi(String userId, RestoreUserRequest requestBody) throws IOException {
+        HttpStatus status;
+        String authHeader = idamTokenGenerator.getIdamAuthorizationHeader();
+        try (Response response = idamClient.restoreUser(authHeader, userId, requestBody)) {
+            status = HttpStatus.valueOf(response.status());
+            if (status == HttpStatus.CONFLICT) {
```

Comment:
You could break down the logic into separate methods, one good candidate would be to create handleConflictStatus() method.